File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/event/AppEventSink.java
package com.adadapted.android.sdk.core.event;

import com.adadapted.android.sdk.core.device.DeviceInfo;

import java.util.Set;

public interface AppEventSink {
    void generateWrappers(DeviceInfo deviceInfo);
    void publishEvent(Set<AppEvent> events);
    void publishError(Set<AppError> errors);
}

File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/event/AppEventSink.kt
package com.adadapted.android.sdk.core.event

import com.adadapted.android.sdk.core.device.DeviceInfo

interface AppEventSink {
    fun generateWrappers(deviceInfo: DeviceInfo)
    fun publishEvent(events: Set<AppEvent>)
    fun publishError(errors: Set<AppError>)
}